ssm_jax/nlgssm/models.py

```diff
@@ -0,0 +1,148 @@
+from jax import numpy as jnp
+from jax import random as jr
+from jax import lax, vmap
+from jax.tree_util import register_pytree_node_class, tree_map
+
+from distrax import MultivariateNormalFullCovariance as MVN
+
+from ssm_jax.ekf.inference import extended_kalman_filter
+from ssm_jax.utils import PSDToRealBijector
+
+@register_pytree_node_class
+class NonLinearGaussianSSM:
+    '''
+    Non-Linear Gaussian State Space Model is defined as follows:
+    p(z_t | z_{t-1}, u_t) = N(z_t | f(z_{t-1}, u_t), Q_t)
+    p(y_t | z_t) = N(y_t | h(z_t, u_t), R_t)
+    p(z_1) = N(z_1 | mu_{1|0}, Sigma_{1|0})
+    where z_t = hidden, y_t = observed, u_t = inputs,
+    dynamics_function = f
+    dynamics_covariance = Q
+    emission_function = h
+    emissions_covariance = R
+    initial_mean = mu_{1|0}
+    initial_covariance = Sigma_{1|0}
+    Optional parameters (default to 0)
+    
+    TODO: Add additional parameters for non-EKF algorithms 
```

Comment:
I think of the hparams (alpha, beta, kappa) as algorithm specific, not part of the model.